### PR TITLE
doc: add role management

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related to https://github.com/harvester/harvester/issues/4786

This is the document part for various roles supported.

### CODE REVIEW

File: docs/host/host.md [hunk 1]
```diff
@@ -58,6 +58,24 @@ Once resolved, the last step can be skipped.
 
 ![delete.png](/img/v1.2/host/delete-node.png)
 
+## Roles Management
+
+It's very common for users to need to replace the existing management node due to hardware issues. However, in previous Harvester versions, it was not easy to accurately promote a specific worker node to a management node.
+
+To make it more flexible and easier, Harvester introduces various roles in v1.3.0. The roles are the following:
```
Comment (jillian-maroket) [inline docs/host/host.md:65]:
```suggestion
## Role Management

Hardware issues may force you to replace the management node. In earlier Harvester versions, accurately promoting a specific worker node to a management node was not easy. Harvester v1.3.0 improves the process by introducing the following roles:
```

File: docs/host/host.md [hunk 2]
```diff
@@ -58,6 +58,24 @@ Once resolved, the last step can be skipped.
...
+
+- **Management**: If the node is a management role, it is highly prioritized to be promoted to the management nodes.
+- **Worker**: If the node is a worker role, it will never be promoted to the management nodes.
+- **Witness**: If the node is a witness role, it will only be an etcd node in this cluster.
```
Comment (jillian-maroket) [inline docs/host/host.md:69]:
```suggestion
- **Management**: Allows a node to be prioritized when Harvester promotes nodes to management nodes.
- **Witness**: Restricts a node to being a witness node (only functions as an etcd node) in a specific cluster.
- **Worker**: Restricts a node to being a worker node (never promoted to management node) in a specific cluster.
```

File: docs/host/host.md [hunk 3]
```diff
@@ -58,6 +58,24 @@ Once resolved, the last step can be skipped.
...
+
+:::caution
+
+Only one witness node is allowed in the cluster for now.
+
+:::
+
+Please check the installation guide [here](/v1.3/install/index) to see how to set the roles during the installation.
```
Comment (jillian-maroket) [inline docs/host/host.md:77]:
```suggestion
:::caution
Harvester currently allows only one witness node in the cluster.
:::

For more information about assigning roles to nodes, see [ISO Installation](/v1.3/install/index).
```

File: docs/install/iso-install.md [hunk 4]
```diff
@@ -50,6 +50,15 @@ The following [video](https://youtu.be/X0VIGZ_lExQ) shows a quick overview of an
...
 	:::
 
+1. Choose the role for this node (only allow when Join an existing Harvester Cluster)
+
+	![choose-node-role.png](/img/v1.3/install/select-role.png)
+
+    - `Default Role`: The default role allows this node to be a management node or a worker node. This role does not have any specific promotion policy.
+	- `Management Role`: The management role gives the higher priority for promotion to be a management node.
+	- `Witness Role`: The witness role restricts this node to being a witness node (acting as an etcd node).
+	- `Worker Role`: The worker role will restrict this node to only be a worker node.
```
Comment (jillian-maroket) [inline docs/install/iso-install.md:60]:
```suggestion
1. Choose a role for the node. You are required to perform this step if you selected the installation mode `Join an existing Harvester cluster`.

  ![choose-node-role.png](/img/v1.3/install/select-role.png)

  - `Default Role`: Allows a node to function as a management node or a worker node. This role does not have any specific privileges or restrictions.
  - `Management Role`: Allows a node to be prioritized when Harvester promotes nodes to management nodes.
  - `Witness Role`: Restricts a node to being a witness node (only functions as an etcd node) in a specific cluster.
  - `Worker Role`: Restricts a node to being a worker node (never promoted to management node) in a specific cluster.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Comment by Vicente-Cheng** [comment]
> Let me know if you have concerns about the changes.

Updated! Thanks for the review!

**Comment by Vicente-Cheng** [comment]
Hi @jillian-maroket, 
Sorry, I added some info on the pxe-boot page.
Could you help to review it again?

File: docs/install/harvester-configuration.md [hunk 5]
```diff
@@ -441,6 +441,19 @@ install:
...
+### `install.role`
+
+#### Definition
+
+Harvester installation role:
+
+- `Default`: Allows a node to function as a management node or a worker node.
+- `Management`: Allows a node to be prioritized when Harvester promotes nodes to management nodes.
+- `Worker`: Restricts a node to being a worker node (never promoted to management node) in a specific cluster.
+- `Witness`: Restricts a node to being a witness node (only functions as an etcd node) in a specific cluster.
```
Comment (bk201) [inline docs/install/harvester-configuration.md:453]:
```suggestion
- `default`: Allows a node to function as a management node or a worker node.
- `management`: Allows a node to be prioritized when Harvester promotes nodes to management nodes.
- `worker`: Restricts a node to being a worker node (never promoted to management node) in a specific cluster.
- `witness`: Restricts a node to being a witness node (only functions as an etcd node) in a specific cluster.
```

File: docs/install/harvester-configuration.md [hunk 6]
```diff
@@ -441,6 +441,19 @@ install:
...
+
+If you don't specify the role, the `Default` role will be used.
```
Comment (bk201) [inline docs/install/harvester-configuration.md:455]:
```suggestion
If you don't specify the role, the `default` role will be used.
```

**Review by bk201 — APPROVED** [review]
Thanks!

File: docs/install/harvester-configuration.md [hunk 7]
```diff
@@ -441,6 +441,19 @@ install:
...
+#### Definition
+
+Harvester installation role:
+
+- `default`: Allows a node to function as a management node or a worker node.
+- `management`: Allows a node to be prioritized when Harvester promotes nodes to management nodes.
+- `worker`: Restricts a node to being a worker node (never promoted to management node) in a specific cluster.
+- `witness`: Restricts a node to being a witness node (only functions as an etcd node) in a specific cluster.
+
+If you don't specify the role, the `default` role will be used.
```
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply docs/install/harvester-configuration.md]: removed

**Pull request merged**